Manatee.Json/Serialization/AbstractionMap.cs

```diff
@@ -94,17 +94,22 @@ public void RemoveMap<TAbstract>(bool removeRelated = true)
 		/// <returns>The mapped type if a mapping exists; otherwise the abstraction type.</returns>
 		public Type GetMap(Type type)
 		{
-			if (!type.GetTypeInfo().IsAbstract && !type.GetTypeInfo().IsInterface) return type;
 			if (_registry.TryGetValue(type, out Type tConcrete)) return tConcrete;
-
+			if (!type.GetTypeInfo().IsAbstract && !type.GetTypeInfo().IsInterface)
```

Comment:
@sixlettervariables, this is breaking two tests: 

- `JsonSerializerTests.MapGenericAbstraction_BaseClass_Success()`
- `JsonSerializerTests.MapGenericAbstraction_WithOverride_Success()`

The purpose of these two tests is to verify that a non-abstract base class registration can be mapped to a derived class.  The generic type has to be checked before we can add it to the registry.

Reply:
I've moved it to after the generic check.